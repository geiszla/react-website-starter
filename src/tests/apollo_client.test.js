import { InMemoryCache } from 'apollo-cache-inmemory';
import fetch from 'node-fetch';

import apolloClient from '../apollo_client';

global.fetch = fetch;
global.SERVER_URL = 'abc';

describe('Apollo Client', () => {
  it('should export a valid Apollo Client object without SSR', () => {
    expect(apolloClient(false, undefined, new InMemoryCache())).toMatchSnapshot();
  });

  it('should export a valid Apollo Client object with SSR', () => {
    expect(apolloClient(true, undefined, new InMemoryCache())).toMatchSnapshot();
  });
});
