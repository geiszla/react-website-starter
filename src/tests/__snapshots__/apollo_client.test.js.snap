// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apollo Client should export a valid Apollo Client object with SSR 1`] = `
ApolloClient {
  "cache": InMemoryCache {
    "addTypename": true,
    "cacheKeyRoot": CacheKeyNode {
      "children": null,
      "key": null,
    },
    "config": Object {
      "addTypename": true,
      "dataIdFromObject": [Function],
      "fragmentMatcher": HeuristicFragmentMatcher {},
    },
    "data": DepTrackingCache {
      "data": Object {},
      "depend": [Function],
    },
    "maybeBroadcastWatch": [Function],
    "optimistic": Array [],
    "silenceBroadcast": false,
    "storeReader": StoreReader {
      "cacheKeyRoot": CacheKeyNode {
        "children": null,
        "key": null,
      },
      "executeSelectionSet": [Function],
      "executeStoreQuery": [Function],
      "keyMaker": QueryKeyMaker {
        "cacheKeyRoot": CacheKeyNode {
          "children": null,
          "key": null,
        },
        "perQueryKeyMakers": Map {},
      },
    },
    "storeWriter": StoreWriter {},
    "typenameDocumentCache": Map {},
    "watches": Set {},
  },
  "defaultOptions": Object {},
  "disableNetworkFetches": true,
  "link": ApolloLink {
    "request": [Function],
  },
  "mutate": [Function],
  "query": [Function],
  "queryDeduplication": true,
  "reFetchObservableQueries": [Function],
  "resetStore": [Function],
  "resetStoreCallbacks": Array [],
  "ssrMode": true,
  "store": DataStore {
    "cache": InMemoryCache {
      "addTypename": true,
      "cacheKeyRoot": CacheKeyNode {
        "children": null,
        "key": null,
      },
      "config": Object {
        "addTypename": true,
        "dataIdFromObject": [Function],
        "fragmentMatcher": HeuristicFragmentMatcher {},
      },
      "data": DepTrackingCache {
        "data": Object {},
        "depend": [Function],
      },
      "maybeBroadcastWatch": [Function],
      "optimistic": Array [],
      "silenceBroadcast": false,
      "storeReader": StoreReader {
        "cacheKeyRoot": CacheKeyNode {
          "children": null,
          "key": null,
        },
        "executeSelectionSet": [Function],
        "executeStoreQuery": [Function],
        "keyMaker": QueryKeyMaker {
          "cacheKeyRoot": CacheKeyNode {
            "children": null,
            "key": null,
          },
          "perQueryKeyMakers": Map {},
        },
      },
      "storeWriter": StoreWriter {},
      "typenameDocumentCache": Map {},
      "watches": Set {},
    },
  },
  "version": "2.4.2",
  "watchQuery": [Function],
}
`;

exports[`Apollo Client should export a valid Apollo Client object without SSR 1`] = `
ApolloClient {
  "cache": InMemoryCache {
    "addTypename": true,
    "cacheKeyRoot": CacheKeyNode {
      "children": null,
      "key": null,
    },
    "config": Object {
      "addTypename": true,
      "dataIdFromObject": [Function],
      "fragmentMatcher": HeuristicFragmentMatcher {},
    },
    "data": DepTrackingCache {
      "data": Object {},
      "depend": [Function],
    },
    "maybeBroadcastWatch": [Function],
    "optimistic": Array [],
    "silenceBroadcast": false,
    "storeReader": StoreReader {
      "cacheKeyRoot": CacheKeyNode {
        "children": null,
        "key": null,
      },
      "executeSelectionSet": [Function],
      "executeStoreQuery": [Function],
      "keyMaker": QueryKeyMaker {
        "cacheKeyRoot": CacheKeyNode {
          "children": null,
          "key": null,
        },
        "perQueryKeyMakers": Map {},
      },
    },
    "storeWriter": StoreWriter {},
    "typenameDocumentCache": Map {},
    "watches": Set {},
  },
  "defaultOptions": Object {},
  "disableNetworkFetches": false,
  "link": ApolloLink {
    "request": [Function],
  },
  "mutate": [Function],
  "query": [Function],
  "queryDeduplication": true,
  "reFetchObservableQueries": [Function],
  "resetStore": [Function],
  "resetStoreCallbacks": Array [],
  "ssrMode": false,
  "store": DataStore {
    "cache": InMemoryCache {
      "addTypename": true,
      "cacheKeyRoot": CacheKeyNode {
        "children": null,
        "key": null,
      },
      "config": Object {
        "addTypename": true,
        "dataIdFromObject": [Function],
        "fragmentMatcher": HeuristicFragmentMatcher {},
      },
      "data": DepTrackingCache {
        "data": Object {},
        "depend": [Function],
      },
      "maybeBroadcastWatch": [Function],
      "optimistic": Array [],
      "silenceBroadcast": false,
      "storeReader": StoreReader {
        "cacheKeyRoot": CacheKeyNode {
          "children": null,
          "key": null,
        },
        "executeSelectionSet": [Function],
        "executeStoreQuery": [Function],
        "keyMaker": QueryKeyMaker {
          "cacheKeyRoot": CacheKeyNode {
            "children": null,
            "key": null,
          },
          "perQueryKeyMakers": Map {},
        },
      },
      "storeWriter": StoreWriter {},
      "typenameDocumentCache": Map {},
      "watches": Set {},
    },
  },
  "version": "2.4.2",
  "watchQuery": [Function],
}
`;
